### ISSUE TITLE
When include -e options, should set tag_file_name to `TAGS' automaticlly.

### ISSUE THREAD

**Issue by zw963** [tmm1/ripper-tags#56]
some emacs package auto search the correct directory which `TAGS` should
live in, and auto add `-f /my/tags/file/path/default_tags_name`.

 If we hardcode with `-f ./TAGS`, will make this(auto search) function invalid.

So, I have to do is when update this gem, chanage `tags_file_name` default value
to `./TAGS` manually to make emacs package to worked.

following is my emacs config:

```cl
(require 'ctags-update)
(setq ctags-update-command "ripper-tags"
      ctags-update-delay-seconds 30

      ;; Here we could not specify -f arg. because it will broken this package.
      ctags-update-other-options '("--exclude='*.elc'" "--exclude='*.class'" "--exclude='.git'"
                                   "--exclude='.svn'" "--exclude='SCCS'" "--exclude='RCS'"
                                   "--exclude='CVS'" "--exclude='EIFGEN'" "--exclude='vendor'"
                                   "--force" "--extra=q" "-e")
      )
```

**Comment by tmm1** [comment]
Can you open a pull request with a fix?

**Comment by zw963** [comment]
Is there need add more test for this?
Could not make test to work well for me. (tested with 1.9.3 and 2.2.3)

1.9.3
```sh
Run options: 

# Running tests:

..............Usage: rake_test_loader [options] FILES...
rake aborted!
Command failed with status (1): [ruby -w -I"lib" -I"/home/zw963/.rvm/gems/ruby-1.9.3-p551@ripper-tags/gems/rake-11.2.2/lib" "/home/zw963/.rvm/gems/ruby-1.9.3-p551@ripper-tags/gems/rake-11.2.2/lib/rake/rake_test_loader.rb" "test/test_cli.rb" "test/test_data_reader.rb" "test/test_formatters.rb" "test/test_ripper_tags.rb" ]
/home/zw963/.rvm/gems/ruby-1.9.3-p551@ripper-tags/gems/rake-11.2.2/exe/rake:27:in `<top (required)>'
/home/zw963/.rvm/gems/ruby-1.9.3-p551@ripper-tags/bin/ruby_executable_hooks:15:in `eval'
/home/zw963/.rvm/gems/ruby-1.9.3-p551@ripper-tags/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => test
(See full trace by running task with --trace)
```

2.2.3
```sh
/home/zw963/Git/ripper-tags/test/test_cli.rb:1:in `require': cannot load such file -- test/unit (LoadError)
	from /home/zw963/Git/ripper-tags/test/test_cli.rb:1:in `<top (required)>'
	from /home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/lib/rake/rake_test_loader.rb:15:in `require'
	from /home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/lib/rake/rake_test_loader.rb:15:in `block in <main>'
	from /home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/lib/rake/rake_test_loader.rb:4:in `select'
	from /home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/lib/rake/rake_test_loader.rb:4:in `<main>'
rake aborted!
Command failed with status (1): [ruby -w -I"lib" -I"/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/lib" "/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/lib/rake/rake_test_loader.rb" "test/test_cli.rb" "test/test_data_reader.rb" "test/test_formatters.rb" "test/test_ripper_tags.rb" ]
/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/exe/rake:27:in `<top (required)>'
/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/bin/ruby_executable_hooks:15:in `eval'
/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => test
(See full trace by running task with --trace)
```

**Comment by zw963** [comment]
@tmm1 , Add a new PR #57  for this, tough no test.

**Comment by zw963** [comment]
2.2.3 if install test-unit  3.1.9

```sh
Loaded suite /home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/lib/rake/rake_test_loader
Started
..............Usage: rake_test_loader [options] FILES...


Finished in 0.008489532 seconds.
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
14 tests, 19 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
1649.09 tests/s, 2238.05 assertions/s
rake aborted!
Command failed with status (1): [ruby -w -I"lib" -I"/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/lib" "/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/lib/rake/rake_test_loader.rb" "test/test_cli.rb" "test/test_data_reader.rb" "test/test_formatters.rb" "test/test_ripper_tags.rb" ]
/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/exe/rake:27:in `<top (required)>'
/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/bin/ruby_executable_hooks:15:in `eval'
/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => test
(See full trace by running task with --trace)
```

**Comment by mislav** [comment]
Sorry that you had trouble. I've just added test-unit to our gem bundle so that people don't have to install the test-unit gem manually. Please see the latest master branch:

```
$ bundle exec rake

...
58 tests, 129 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
```

**Comment by mislav** [comment]
@zw963 I've found the problem you were having with the test suite prematurely exiting with the message `Usage: rake_test_loader …`. I've fixed it in d5cdd728a30e3c553f065c3cee2ef8eea2514fc5.

**Comment by zw963** [comment]
> I've found the problem you were having with the test suite prematurely exiting with the message Usage: rake_test_loader …. I've fixed it in d5cdd72.

Cool, Test start to work for me now, at least, when test failed, I could see the failed result now.

Though, two test always failed on my local machine, seem like just `array equality` things.
maybe you could reproduce with ruby 2.2.3. 

```cl
Loaded suite /home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/lib/rake/rake_test_loader
Started
....................F
=============================================================================================================================================================================================
Failure: test_file_finder(DataReaderTest)
/home/zw963/Git/ripper-tags/test/test_data_reader.rb:44:in `test_file_finder'
     41:       very/deep/script.rb
     42:       very/inter.rb
     43:     ]
  => 44:     assert_equal expected, files
     45:   end
     46: 
     47:   def test_file_finder_no_exclude
<["encoding.rb", "very/deep/script.rb", "very/inter.rb"]> expected but was
<["very/deep/script.rb", "very/inter.rb", "encoding.rb"]>

diff:
? ["encoding.rb", "very/deep/script.rb", "very/inter.rb", "encoding.rb"]
=============================================================================================================================================================================================
..F
=============================================================================================================================================================================================
Failure: test_file_finder_exclude_glob(DataReaderTest)
/home/zw963/Git/ripper-tags/test/test_data_reader.rb:64:in `test_file_finder_exclude_glob'
     61:       encoding.rb
     62:       very/inter.rb
     63:     ]
  => 64:     assert_equal expected, files
     65:   end
     66: 
     67:   def test_file_finder_cleanpath
<["encoding.rb", "very/inter.rb"]> expected but was
<["very/inter.rb", "encoding.rb"]>

diff:
? ["encoding.rb", "very/inter.rb", "encoding.rb"]
=============================================================================================================================================================================================
......................................

Finished in 0.025170509 seconds.
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
62 tests, 137 assertions, 2 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
96.7742% passed
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
2463.20 tests/s, 5442.88 assertions/s
rake aborted!
Command failed with status (1): [ruby -w -I"lib" -I"/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/lib" "/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/lib/rake/rake_test_loader.rb" "test/test_cli.rb" "test/test_data_reader.rb" "test/test_formatters.rb" "test/test_ripper_tags.rb" ]
/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/gems/rake-11.2.2/exe/rake:27:in `<top (required)>'
/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/bin/ruby_executable_hooks:15:in `eval'
/home/zw963/.rvm/gems/ruby-2.2.3@ripper-tags/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => test
(See full trace by running task with --trace)
```
